### PR TITLE
beauties: init at 0.0.14

### PR DESCRIPTION
**Pull request by RaitoBezarius**
##### Motivation for this change

This open the way for a beauties NixOS service, which would enable a little nice web application for minimalist personal services.

This is my first contribution, I'm not sure how to compute the `vendorSha256`, but will do ASAP if possible and will update what I tried.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/web-apps/beauties/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  preBuild = ''
+    	go generate github.com/dsx/beauties/cmd/...
+  '';
+
+  src = fetchFromGitHub {
+    owner = "dsx";
+    repo = pname;
+    rev = "e27e7f5ce69afe7ce966edd568d9c2ec219f7864";
+    sha256 = "1rzi4x8ik4fiwmwfff2pqsb7nsvjncwbrikqnzzqi9dv4jwraymh";
+  };
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/beauties/default.nix:23]:
```suggestion
  src = fetchFromGitHub {
    owner = "dsx";
    repo = pname;
    rev = "e27e7f5ce69afe7ce966edd568d9c2ec219f7864";
    sha256 = "1rzi4x8ik4fiwmwfff2pqsb7nsvjncwbrikqnzzqi9dv4jwraymh";
  };

  vendorSha256 = null;

  nativeBuildInputs = [ go-bindata ];

  subPackages = ["cmd/beauties"];

  preBuild = ''
    go generate github.com/dsx/beauties/cmd/...
  '';

  doCheck = false;
```

File: pkgs/servers/web-apps/beauties/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  meta = with stdenv.lib; {
+    description = "Essential personal Internet services (pastebin, file upload, etc.)";
+    homepage = "https://github.com/dsx/beauties";
+    license = licenses.mit;
+    maintainers = with maintainers; [];
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/beauties/default.nix:29]: Please add yourself as a maintainer.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -1937,6 +1937,8 @@ in
 
   fileshelter = callPackage ../servers/web-apps/fileshelter { };
 
+  beauties = callPackage ../servers/web-apps/beauties { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:1940]: Please sort this.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 97476` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>beauties</li>
  </ul>
</details>

File: pkgs/servers/web-apps/beauties/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  vendorSha256 = null;
+  doCheck = false;
+
+  nativeBuildInputs = [ go-bindata ];
+
+  subPackages = ["cmd/beauties"];
+
+  preBuild = ''
+    	go generate github.com/dsx/beauties/cmd/...
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/beauties/default.nix:15]:
```suggestion
    go generate github.com/dsx/beauties/cmd/...
```

File: pkgs/servers/web-apps/beauties/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  '';
+
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/beauties/default.nix:18]: Please move this after version.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by RaitoBezarius** [comment]
closing as I am not using this anymore.

**Pull request closed without merging**